### ISSUE TITLE
[NLL] Spurious warning in the last Nightly

### ISSUE THREAD

**Issue by leonardo-m** [rust-lang/rust#50410]
```
#![feature(nll)]
fn main() {
    let _ = [1].iter().map(|&x| x);
}
```

Gives a spurious warning:

```
warning: variable does not need to be mutable
 --> ...\test.rs:3:29
  |
3 |     let _ = [1].iter().map(|&x| x);
  |                             ^^ help: remove this `mut`
  |
  = note: #[warn(unused_mut)] on by default
```

If I remove the nll feature the warning goes away.

```
rustc 1.27.0-nightly (8a37c75a3 2018-05-02)
binary: rustc
commit-hash: 8a37c75a3a661385cc607d934c70e86a9eaf5fd7
commit-date: 2018-05-02
host: x86_64-pc-windows-gnu
release: 1.27.0-nightly
LLVM version: 6.0
```

**Comment by Robbepop** [comment]
This is a duplicate of https://github.com/rust-lang/rust/issues/50343
